### PR TITLE
Move adstock and saturation method imports to mmm.__all__

### LINKED ISSUE #892: import `adstock|saturation_from_dict` & `register_adstock|saturation_transformation` in `pymc_marketing.mmm`
Before: 
```python
from pymc_marketing.mmm.components.adstock import adstock_from_dict, register_adstock_transformation
from pymc_marketing.mmm.components.saturation import saturation_from_dict, register_saturation_transformation
```
After: 
```python
from pymc_marketing.mmm import (
    adstock_from_dict, 
    saturation_from_dict, 
    register_adstock_transformation, 
    register_saturation_transformation, 
)
```

### PR DESCRIPTION
**Pull request by PabloRoque**
## Description

* Adds `adstock|saturation_from_dict` to `mmm.__all__`
* Adds `register_adstock|saturation_transformation` to `mmm.__all__`
* chore: move imports in tests
* chore: add `pydantic`, `preliz` and `pytest-mock` to `environment.yml` to match dependencies in `project.toml` and run tests

## Related Issue

- [x] Closes #892 
- [ ] Related to #

## Checklist

- [x] Checked that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html)
- [x] Included tests that prove the fix is effective or that the new feature works
- [ ] Added necessary documentation (docstrings and/or example notebooks)
- [ ] If you are a pro: each commit corresponds to a [relevant logical change](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes)

## Modules affected

- [x] MMM
- [ ] CLV

## Type of change

- [ ] New feature / enhancement
- [ ] Bug fix
- [ ] Documentation
- [x] Maintenance
- [ ] Other (please specify):

----
📚 Documentation preview 📚: https://pymc-marketing--908.org.readthedocs.build/en/908/

### CODE REVIEW

**Comment by wd60622** [comment]
Hi @PabloRoque. Thank you for taking this issue!

The UML failure unrelated to these changes. While I have you here, could you make a change to the UML.yaml file and give this a try: 
```yaml
# .github/workflows/uml.yml
# At the bottom of the file in the last step
      - name: Push up the changes
        run: |
          git add docs/source/_static/*.png
          if git diff --staged --exit-code; then
            echo "No changes to commit"
          else
            echo "Committing the changes"
            git commit -m "Update UML Diagrams"
            git push origin HEAD:${GITHUB_HEAD_REF}
          fi
```

File: .github/workflows/uml.yml [hunk 1]
```diff
@@ -35,10 +35,11 @@ jobs:
 
       - name: Push up the changes
         run: |
-          git add docs/source/uml/*.png
+          git add docs/source/_static/*.png
```
Comment (wd60622) [inline .github/workflows/uml.yml:38]: Shoot. Not this change. 
Sorry copy and paste
```suggestion
          git add docs/source/uml/*.png
```

**Comment by PabloRoque** [comment]
@wd60622 Gave it a go to fix the GHA as per your instruction. 

Thanks promptly jumping into this PR.

File: .github/workflows/uml.yml [hunk 2]
```diff
@@ -39,6 +39,7 @@ jobs:
           if git diff --staged --exit-code; then
             echo "No changes to commit"
           else
+            echo "Committing the changes"
             git commit -m "Update UML Diagrams"
-            git push
+            git push origin HEAD:${GITHUB_HEAD_REF}
           fi
```
Comment (wd60622) [inline .github/workflows/uml.yml:45]:
```suggestion
          fi
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

**Comment by wd60622** [comment]
Sorry. using your PR as some testing grounds 😆

**Comment by wd60622** [comment]
One more trial change for the uml.yaml file

Could you add: 
```yaml
permissions: write-all
```
This is be at root level. i.e. the same level as "name", "on", "jobs"

**Review by wd60622 — APPROVED** [review]
Thanks for making these edits (and helping with these permission issues. We can merge these in and try later)

Looks good!

**Pull request merged**